third_party/move/move-vm/runtime/src/tracing.rs

```diff
@@ -69,16 +67,10 @@ pub(crate) fn trace(
 ) {
     if *TRACING_ENABLED {
         let f = &mut *LOGGING_FILE.lock().unwrap();
-        writeln!(
-            f,
-            "{}-{:?},{},{},{:?}",
-            process::id(),
-            thread::current().id(),
-            function_desc.pretty_string(),
-            pc,
-            instr,
-        )
-        .unwrap();
+        let mut buf_writer = std::io::BufWriter::new(f);
+        buf_writer
```

Comment:
https://doc.rust-lang.org/std/io/struct.BufWriter.html# says:

> "It is critical to call [flush](https://doc.rust-lang.org/std/io/struct.BufWriter.html#method.flush) before BufWriter<W> is dropped. Though dropping will attempt to [flush](https://doc.rust-lang.org/std/io/struct.BufWriter.html#method.flush) the contents of the buffer, any errors that happen in the process of dropping will be ignored. Calling flush ensures that the buffer is empty and thus dropping will not even attempt file operations."

This whole approach seems a bit questionable.  Putting the BufWriter inside LOGGING_FILE would make more sense.  I'd also try using with_capacity() instead of new(), so you can specify some larger capacity, at least 16K (16384).  The default seems to be 8K.  Much larger may still be good, even 4MiB.  Not sure how to ensure that the global gets flushed, though.  Do we have an exit handler somewhere?

Reply:
Thanks for the review, @brmataptos. As you suggested, I move this to `LOGGING_FILE_WRITER` above.